### PR TITLE
[FIX] account: manual import of matching numbers

### PR DESCRIPTION
**Pull request by william-andre**
The import feature of matching numbers was done manually in all imports, but the generic import was missing.

For instance, with this file:
```csv
name,line_ids/account_id,line_ids/debit,line_ids/credit,line_ids/matching_number
test 2,400000,,121,1
,500000,121
test 1,400000,121,,1
,451000,,21
,700000,,100
```

The system wouldn't understand that this is an imported number.

### CODE REVIEW

File: addons/account/models/account_move_line.py [hunk 1]
```diff
@@ -2631,6 +2640,7 @@ def _prepare_exchange_difference_move_vals(self, amounts_list, company=None, exc
                     'debit': -amount_residual if amount_residual < 0.0 else 0.0,
                     'credit': amount_residual if amount_residual > 0.0 else 0.0,
                     'amount_currency': -amount_residual_currency,
+                    'full_reconcile_id': line.full_reconcile_id.id,
```
Comment (h4818) [inline addons/account/models/account_move_line.py:2643]: This doesn't seem to be related to the import of matching numbers....
shouldn't there be a test to check the matching numbers of all reconciled lines including the exchange diff

**Review by h4818 — COMMENTED** [review]
Mostly good, just wondering about the exchange diff change

File: addons/account/tests/test_account_move_reconcile.py [hunk 2]
```diff
@@ -4804,8 +4804,8 @@ def test_reconcile_import(self):
         line_4 = self.create_line_for_reconciliation(-500.0, -500.0, comp_curr, '2016-01-01')
         line_4.move_id.button_draft()
         line_5 = self.create_line_for_reconciliation(200.0, 200.0, comp_curr, '2016-01-01')
-        (line_1 + line_2 + line_3).matching_number = 'I11111'
-        (line_4 + line_5).matching_number = 'I22222'
+        (line_1 + line_2 + line_3).matching_number = '11111'  # Will be converted to a temporary number
+        (line_4 + line_5).matching_number = '22222'  # Will be converted to a temporary number
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/tests/test_account_move_reconcile.py:4808]: Having to hardcode the `I` in the test in the first place was annoying me.
I don't want the test to be implementation dependent, only have the behavior working as expected for the end user

File: addons/account/models/account_full_reconcile.py [hunk 3]
```diff
@@ -38,6 +38,6 @@ def unlink(self):
     @api.model_create_multi
     def create(self, vals_list):
         fulls = super().create(vals_list)
-        for full in fulls:
+        for full in fulls.with_context(update_matching_number=True):
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_full_reconcile.py:41]: I don't understand your suggestions, we are entering the context where we are "updating the matching number programmatically", as opposed to "setting the matching number manually"; not fixing or checking anything

File: addons/account/models/account_move_line.py (hunk 1 above)
Comment (william-andre) [reply addons/account/models/account_move_line.py:2643]: I added this line because of a failing test (see the first batch)
Basically, it is adding a exchange difference on the full reconcile after it was created; but this is only rarely done.
Basically that failing test is now checking what you asked too, thanks to the constraint :laughing:

File: addons/account/tests/test_account_move_reconcile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (h4818) [reply addons/account/tests/test_account_move_reconcile.py:4808]: ok

File: addons/account/models/account_full_reconcile.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (h4818) [reply addons/account/models/account_full_reconcile.py:41]: my mistake, I read it incorrectly... 
then a better name would be `skip_matching_number_check`

[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_full_reconcile.py]: Done!
